packages/deployer/internal/storage-ast-validator.js

```diff
@@ -39,25 +39,63 @@ class ModuleStorageASTValidator {
     for (var [contractName, ast] of Object.entries(this.asts)) {
       const slots = findYulStorageSlotAssignments(contractName, ast);
 
-      slots.forEach((position) => namespaces.push({ contractName, position }));
+      slots.forEach((slot) => namespaces.push({ contractName, slot }));
     }
 
     const duplicates = this.findDuplicateNamespaces(namespaces);
 
     const errors = [];
     if (duplicates) {
       const details = duplicates.map(
-        (d) => `  > ${d.position} found in storage contracts ${d.contracts}\n`
+        (d) => `  > ${d.slot} found in storage contracts ${d.contracts}\n`
       );
 
       errors.push({
-        msg: `Duplicate namespaces found!\n${details.join('')}`,
+        msg: `Duplicate namespaces slot found!\n${details.join('')}`,
       });
     }
 
     return errors;
   }
 
+  findNamespaceSlotChanges() {
```

Comment:
This new function will check the new namespace slots against the previous one to detect any deletion or change in the slot address.